site/src/pages/TerminalPage/TerminalPage.tsx

```diff
@@ -134,13 +132,21 @@ const TerminalPage: FC = () => {
         handleWebLinkRef.current(uri);
       }),
     );
-    setTerminal(terminal);
+
     terminal.open(xtermRef.current);
-    const listener = () => {
-      // This will trigger a resize event on the terminal.
-      fitAddon.fit();
-    };
+
+    // We have to fit twice here. It's unknown why, but the first fit will
+    // overflow slightly in some scenarios. Applying a second fit resolves this.
+    fitAddon.fit();
+    fitAddon.fit();
```

Comment:
I don't know much about xterm, but moving the calls does seem like a good change.

Just making sure for my own understanding – is the terminal expected to resize only when the browser/window viewport does?